scripts/run_scrnaseq.R

```diff
@@ -0,0 +1,382 @@
+# Rscript scripts/run_scrnaseq.R --project-id=pbmc --path-data-csv=test_datasets/10x_SmartSeq2_pbmc_GSE132044/path_data.csv --path-out=test_datasets/10x_SmartSeq2_pbmc_GSE132044/results --deg-contrasts=test_datasets/10x_SmartSeq2_pbmc_GSE132044/deg_contrasts.xlsx --path-to-git=.
+#!/usr/bin/env Rscript
+
+getwd()
+
+# Get arguments from commandline
+arguments = commandArgs(trailingOnly=FALSE)
+
+# Path to script
+script_index = grep("--file", arguments)
+script_dir = dirname(sub("--file=", "", arguments[script_index]))
+script_dir = normalizePath(script_dir)
+script_name = basename(sub("--file=", "", arguments[script_index]))
+
+# Which packages are needed
+library(knitr)
+suppressPackageStartupMessages(library("argparse"))
+
+# Prepare parser for arguments
+parser = ArgumentParser(
+  add_help=TRUE,
+  prog=script_name,
+  description="Bioinformatics analysis workflow for single-cell RNA-seq analysis. The workflow is based on Seurat, and contains additional visualisations, tables and documentation to better understand the analysis. The workflow supports RNA sequencing data from one or more samples processed with 10X Genomics and SmartSeq-2.",
+  epilog="Upon successful completion, the output directory will contain an HTML report, ....")
+
+# Set mutually exclusive groups
+normalizationGroup = parser$add_mutually_exclusive_group(required=FALSE)
+normalizationGroup$add_argument("--standard-RNA", action="store_true", help="Run with logNormalization and standard parameters", required=FALSE)
+normalizationGroup$add_argument("--standard-SCT", action="store_true", help="Run with SCTransform and standard parameters", required=FALSE)
+
+speciesGroup = parser$add_mutually_exclusive_group(required=FALSE)
+speciesGroup$add_argument("--human", action="store_true", help="Annotation for human", required=FALSE)
+speciesGroup$add_argument("--mouse", action="store_true", help="Annotation for mouse", required=FALSE)
+
+# Get project and sample description
+parser$add_argument("--project-id", action="store", help="Project ID", dest="project_id", required=TRUE)
+
+parser$add_argument("--path-data-csv", action="store", help="CSV file containing the path data of the samples", dest="path_data_csv",required=TRUE)
+
+parser$add_argument("--downsample-cells-n", action="store", type="integer", help="Downsample data to at most N cells (default: #default)", dest="downsample_cells_n", default=0)
```

Comment:
Default should be NULL.

Reply:
@tglomb It currently is NULL if no downsampling is required. Hence that should work.